pkg/chart/chart.go

```diff
@@ -309,35 +312,64 @@ func (c *Chart) InitNetClient(details *Details) (HTTPClient, error) {
 		caCertPool = x509.NewCertPool()
 	}
 
-	// If additionalCA is set, load it
-	customCA := details.Auth.CustomCA
 	namespace := os.Getenv("POD_NAMESPACE")
 	if namespace == "" {
 		namespace = defaultNamespace
 	}
-	if customCA != nil {
-		caCertSecret, err := c.kubeClient.CoreV1().Secrets(namespace).Get(customCA.SecretKeyRef.Name, metav1.GetOptions{})
+
+	// If the app repo url is specified then we'll grab the customCA and
+	// defaultHeaders from there, otherwise default to the current details.Auth
+	// (until that path is removed).
+	// TODO(absoludity): Check whether there is a reason that Details.Auth is a different type
+	// to appRepo.Spec.Auth (though they have identical fields). The following would be simpler
```

Comment:
you are right, those are duplicated. I am fine deleting the struct definition `Details.Auth`

Reply:
Great, I'll do that separately and get it landed, so I can clean up this PR.

Reply:
Done, much cleaner :)